java-checks-test-sources/src/main/java/checks/security/AuthorizationsStrongDecisionsCheck.java

```diff
@@ -0,0 +1,298 @@
+package checks.security;
+
+import java.io.Serializable;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.function.Predicate;
+import org.springframework.security.access.AccessDecisionVoter;
+import org.springframework.security.access.ConfigAttribute;
+import org.springframework.security.access.PermissionEvaluator;
+import org.springframework.security.core.Authentication;
+
+public class AuthorizationsStrongDecisionsCheck {
+
+  // AccessDecisionVoter =================================================
+
+  class WeakNightVoter implements AccessDecisionVoter {
+    @Override
+    public int vote(Authentication authentication, Object object, Collection collection) { // Noncompliant [[sc=16;ec=20]] {{"vote" method should return at least one time ACCESS_DENIED.}}
+      Calendar calendar = Calendar.getInstance();
+      int currentHour = calendar.get(Calendar.HOUR_OF_DAY);
+      if(currentHour >= 8 && currentHour <= 19) {
+        return ACCESS_GRANTED;
```

Comment:
There's false-negatives because we don't support the second form:

1. `return ACCESS_GRANTED;`
2. `return AccessDecisionVoter.ACCESS_GRANTED;`